solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -854,6 +857,14 @@ public static void createClusterZkNodes(SolrZkClient zkClient)
       throws KeeperException, InterruptedException, IOException {
     ZkCmdExecutor cmdExecutor = new ZkCmdExecutor(zkClient.getZkClientTimeout());
     cmdExecutor.ensureExists(ZkStateReader.LIVE_NODES_ZKNODE, zkClient);
+    cmdExecutor.ensureExists(ZkStateReader.NODE_ROLES, zkClient);
+    for (NodeRoles.Role role : NodeRoles.Role.values()) {
+      cmdExecutor.ensureExists(ZkStateReader.NODE_ROLES + "/" + role.roleName, zkClient);
```

Comment:
Fixed all occurrences here to use a method in NodeRoles class that took Role and Mode. However, left out the ClusterAPI references to use that method since I didn't want to constrain the API calls to be able to resolve strictly typed Node, Role values for backcompat reasons. Maybe, we can add another version of those methods that only took strings, not Role/Mode values?